Sprint 3

AleksandrMenshchikov opened the pull request.

AleksandrMenshchikov committed:
modified:   tsconfig.json

AleksandrMenshchikov committed:
modified:   package.json
	new file:   src/app.css
	new file:   src/app.ts
	deleted:    src/components/form-messenger/index.ts
	renamed:    src/components/form-messenger/form-messenger.css -> src/components/form-settings/form-settings.css
	renamed:    src/components/form-messenger/form-messenger.hbs -> src/components/form-settings/form-settings.hbs
	renamed:    src/components/form-messenger/form-messenger.ts -> src/components/form-settings/form-settings.ts
	new file:   src/components/form-settings/index.ts
	renamed:    src/pages/messenger/index.html -> src/index.html
	modified:   src/pages/404/index.css
	deleted:    src/pages/404/index.html
	modified:   src/pages/404/index.ts
	modified:   src/pages/500/index.css
	deleted:    src/pages/500/index.html
	modified:   src/pages/500/index.ts
	deleted:    src/pages/index.css
	deleted:    src/pages/index.html
	modified:   src/pages/login/index.css
	deleted:    src/pages/login/index.html
	modified:   src/pages/login/index.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts
	new file:   src/pages/settings/index.css
	new file:   src/pages/settings/index.hbs
	new file:   src/pages/settings/index.ts
	modified:   src/pages/signup/index.css
	deleted:    src/pages/signup/index.html
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/app.ts
	modified:   src/components/form-login/form-login.css
	modified:   src/components/form-signup/form-signup.css
	new file:   src/core/Route.ts
	new file:   src/core/Router.ts
	modified:   src/pages/login/index.ts

AleksandrMenshchikov committed:
modified:   src/app.ts
	modified:   src/core/Route.ts
	modified:   src/core/Router.ts
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   .husky/pre-commit
	modified:   package.json

AleksandrMenshchikov committed:
new file:   utils/trim.ts

AleksandrMenshchikov committed:
new file:   utils/merge.ts
	new file:   utils/set.ts

AleksandrMenshchikov committed:
new file:   utils/cloneDeep.ts
	new file:   utils/isEqual.ts
	new file:   utils/isPlainObject.ts

AleksandrMenshchikov committed:
new file:   utils/queryStringify.ts

AleksandrMenshchikov committed:
modified:   app.ts
	modified:   package.json
	deleted:    src/app.ts
	new file:   src/components/button-switch-page/button-switch-page.css
	new file:   src/components/button-switch-page/button-switch-page.hbs
	new file:   src/components/button-switch-page/button-swith-page.ts
	new file:   src/components/button-switch-page/index.ts
	modified:   src/components/form-login/form-login.css
	modified:   src/components/form-login/form-login.hbs
	modified:   src/components/form-login/form-login.ts
	modified:   src/components/form-signup/form-signup.hbs
	modified:   src/components/form-signup/form-signup.ts
	modified:   src/core/Route.ts
	modified:   src/core/Router.ts
	renamed:    src/app.css -> src/index.css
	modified:   src/index.html
	new file:   src/index.ts
	modified:   src/pages/404/index.css
	modified:   src/pages/404/index.hbs
	modified:   src/pages/404/index.ts
	modified:   src/pages/500/index.css
	modified:   src/pages/500/index.hbs
	modified:   src/pages/500/index.ts
	modified:   src/pages/login/index.css
	modified:   src/pages/login/index.hbs
	modified:   src/pages/login/index.ts
	modified:   src/pages/settings/index.css
	modified:   src/pages/settings/index.hbs
	modified:   src/pages/signup/index.css
	modified:   src/pages/signup/index.hbs
	modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   app.ts

AleksandrMenshchikov committed:
new file:   src/api/base-api.ts
	new file:   src/api/baseURLs.ts
	new file:   src/api/signup-api.ts
	modified:   src/index.ts
	modified:   utils/HTTPTransport.ts

AleksandrMenshchikov committed:
modified:   src/index.ts

AleksandrMenshchikov committed:
new file:   src/api/auth-api.ts
	modified:   src/api/base-api.ts
	deleted:    src/api/baseURLs.ts
	deleted:    src/api/signup-api.ts
	new file:   src/api/urls.ts
	new file:   src/controllers/signup-controller.ts
	modified:   src/pages/signup/index.ts
	modified:   utils/HTTPTransport.ts
	modified:   utils/isEqual.ts
	modified:   utils/set.ts

AleksandrMenshchikov committed:
new file:   src/components/form-message-error/form-message-error.css
	new file:   src/components/form-message-error/form-message-error.hbs
	new file:   src/components/form-message-error/form-message-error.ts
	new file:   src/components/form-message-error/index.ts
	modified:   src/components/form-signup/form-signup.hbs
	modified:   src/components/form-signup/form-signup.ts
	modified:   src/controllers/signup-controller.ts
	modified:   src/core/Router.ts
	modified:   src/core/Store.ts
	renamed:    src/hoc/connect.ts -> src/core/connect.ts
	new file:   src/hoc/withButton.ts
	new file:   src/hoc/withFormMessageError.ts

AleksandrMenshchikov committed:
modified:   src/pages/signup/index.ts

AleksandrMenshchikov committed:
modified:   src/api/auth-api.ts
	modified:   src/api/urls.ts
	modified:   src/components/form-login/form-login.hbs
	new file:   src/controllers/login-controller.ts
	modified:   src/core/Store.ts
	modified:   src/pages/login/index.ts
	modified:   src/components/form-login/form-login.ts
	modified:   src/components/form-signup/form-signup.ts

AleksandrMenshchikov committed:
modified:   src/components/form-login/form-login.ts
	modified:   src/components/form-signup/form-signup.ts
	modified:   src/core/Router.ts
	modified:   src/index.ts

AleksandrMenshchikov committed:
modified:   src/controllers/login-controller.ts

AleksandrMenshchikov committed:
modified:   src/core/Store.ts

AleksandrMenshchikov committed:
new file:   src/controllers/initial-controller.ts
	modified:   src/controllers/login-controller.ts
	modified:   src/controllers/signup-controller.ts
	modified:   src/index.ts

AleksandrMenshchikov committed:
modified:   src/components/form-settings/form-settings.ts
	new file:   src/components/loader/index.ts
	new file:   src/components/loader/loader.css
	new file:   src/components/loader/loader.hbs
	new file:   src/components/loader/loader.ts
	modified:   src/controllers/initial-controller.ts
	new file:   src/controllers/logout-controller.ts
	modified:   src/index.ts
	modified:   src/pages/messenger/index.ts
	modified:   src/pages/settings/index.ts

AleksandrMenshchikov committed:
modified:   src/controllers/logout-controller.ts

AleksandrMenshchikov committed:
modified:   src/components/form-settings/form-settings.hbs
	modified:   src/components/form-settings/form-settings.ts

AleksandrMenshchikov committed:
modified:   src/components/form-settings/form-settings.css
	modified:   src/components/form-settings/form-settings.hbs
	modified:   src/components/form-settings/form-settings.ts
	new file:   src/components/profile-title/index.ts
	new file:   src/components/profile-title/profile-title.css
	new file:   src/components/profile-title/profile-title.hbs
	new file:   src/components/profile-title/profile-title.ts

AleksandrMenshchikov committed:
modified:   src/components/form-settings/form-settings.ts
	new file:   src/components/profile-input/index.ts
	new file:   src/components/profile-input/profile-input.css
	new file:   src/components/profile-input/profile-input.hbs
	new file:   src/components/profile-input/profile-input.ts
	modified:   src/controllers/initial-controller.ts
	modified:   src/controllers/login-controller.ts
	modified:   src/controllers/signup-controller.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withProfileInputDisplayName.ts
	new file:   src/hoc/withProfileInputEmail.ts
	new file:   src/hoc/withProfileInputFirstName.ts
	new file:   src/hoc/withProfileInputLogin.ts
	new file:   src/hoc/withProfileInputPhone.ts
	new file:   src/hoc/withProfileInputSecondName.ts

AleksandrMenshchikov committed:
modified:   src/components/form-settings/form-settings.ts
	modified:   src/controllers/initial-controller.ts
	modified:   src/controllers/login-controller.ts
	modified:   src/controllers/signup-controller.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withProfileTitle.ts

AleksandrMenshchikov committed:
modified:   src/components/button/button.css
	new file:   src/components/form-avatar/form-avatar.css
	new file:   src/components/form-avatar/form-avatar.hbs
	new file:   src/components/form-avatar/form-avatar.ts
	new file:   src/components/form-avatar/index.ts
	new file:   src/components/input-file/index.ts
	new file:   src/components/input-file/input-file.css
	new file:   src/components/input-file/input-file.hbs
	new file:   src/components/input-file/input-file.ts
	new file:   src/components/label-file/index.ts
	new file:   src/components/label-file/label-file.css
	new file:   src/components/label-file/label-file.hbs
	new file:   src/components/label-file/label-file.ts
	new file:   src/components/modal-profile-avatar-title/index.ts
	new file:   src/components/modal-profile-avatar-title/modal-profile-avatar-title.css
	new file:   src/components/modal-profile-avatar-title/modal-profile-avatar-title.hbs
	new file:   src/components/modal-profile-avatar-title/modal-profile-avatar-title.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withLabelFile.ts
	new file:   src/hoc/withModalProfileAvatarTitle.ts
	modified:   src/pages/settings/index.css
	modified:   src/pages/settings/index.hbs
	modified:   src/pages/settings/index.ts

AleksandrMenshchikov committed:
modified:   src/api/base-api.ts
	modified:   src/api/urls.ts
	new file:   src/api/users-api.ts
	new file:   src/components/form-avatar-error/form-avatar-error.css
	new file:   src/components/form-avatar-error/form-avatar-error.hbs
	new file:   src/components/form-avatar-error/form-avatar-error.ts
	new file:   src/components/form-avatar-error/index.ts
	modified:   src/components/form-avatar/form-avatar.css
	modified:   src/components/form-avatar/form-avatar.hbs
	modified:   src/components/form-avatar/form-avatar.ts
	new file:   src/controllers/avatar-controller.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withFormAvatarError.ts
	modified:   src/pages/settings/index.ts

AleksandrMenshchikov committed:
modified:   src/api/urls.ts
	new file:   src/components/avatar/avatar.css
	new file:   src/components/avatar/avatar.hbs
	new file:   src/components/avatar/avatar.ts
	new file:   src/components/avatar/index.ts
	modified:   src/components/button-avatar/button-avatar.css
	modified:   src/components/button-avatar/button-avatar.hbs
	modified:   src/components/button-avatar/button-avatar.ts
	modified:   src/components/form-avatar/form-avatar.ts
	modified:   src/components/form-settings/form-settings.ts
	modified:   src/components/label-file/label-file.css
	modified:   src/controllers/avatar-controller.ts
	modified:   src/controllers/initial-controller.ts
	modified:   src/controllers/login-controller.ts
	modified:   src/controllers/signup-controller.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withAvatar.ts
	new file:   src/hoc/withButtonFormAvatar.ts
	modified:   src/pages/settings/index.ts

AleksandrMenshchikov committed:
modified:   src/controllers/initial-controller.ts
	modified:   src/core/Store.ts

AleksandrMenshchikov committed:
modified:   src/pages/settings/index.ts

AleksandrMenshchikov committed:
modified:   src/api/urls.ts
	modified:   src/controllers/avatar-controller.ts
	modified:   src/controllers/initial-controller.ts
	modified:   src/controllers/login-controller.ts

AleksandrMenshchikov committed:
modified:   src/components/form-settings/form-settings.ts

AleksandrMenshchikov committed:
modified:   src/pages/settings/index.ts

AleksandrMenshchikov committed:
modified:   src/api/auth-api.ts
	modified:   src/api/base-api.ts
	modified:   src/api/urls.ts
	modified:   src/api/users-api.ts
	modified:   src/components/button/button.css
	modified:   src/components/form-settings/form-settings.ts
	modified:   src/controllers/avatar-controller.ts
	new file:   src/controllers/profile-controller.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withButtonProfile.ts
	modified:   src/pages/settings/index.ts

AleksandrMenshchikov committed:
modified:   src/components/form-settings/form-settings.ts

AleksandrMenshchikov committed:
modified:   src/api/urls.ts
	modified:   src/api/users-api.ts
	modified:   src/components/form-settings/form-settings.css
	modified:   src/components/form-settings/form-settings.hbs
	new file:   src/components/password-error/index.ts
	new file:   src/components/password-error/password-error.css
	new file:   src/components/password-error/password-error.hbs
	new file:   src/components/password-error/password-error.ts
	new file:   src/controllers/password-controller.ts
	modified:   src/pages/settings/index.ts

AleksandrMenshchikov committed:
modified:   src/components/form-settings/form-settings.ts
	modified:   src/controllers/password-controller.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withPasswordError.ts

AleksandrMenshchikov committed:
modified:   src/api/urls.ts
	modified:   src/api/users-api.ts
	modified:   src/controllers/initial-controller.ts
	modified:   src/core/Store.ts

AleksandrMenshchikov committed:
modified:   src/controllers/initial-controller.ts

AleksandrMenshchikov committed:
new file:   src/api/chats-api.ts
	modified:   src/api/urls.ts
	modified:   src/api/users-api.ts
	new file:   src/components/button-search/button-search.css
	new file:   src/components/button-search/button-search.hbs
	new file:   src/components/button-search/button-search.ts
	new file:   src/components/button-search/index.ts
	new file:   src/components/input-search/index.ts
	new file:   src/components/input-search/input-search.css
	new file:   src/components/input-search/input-search.hbs
	new file:   src/components/input-search/input-search.ts
	new file:   src/components/list-members/index.ts
	renamed:    src/components/member/member.css -> src/components/list-members/list-members.css
	new file:   src/components/list-members/list-members.hbs
	new file:   src/components/list-members/list-members.ts
	new file:   src/components/list-users/index.ts
	new file:   src/components/list-users/list-users.css
	new file:   src/components/list-users/list-users.hbs
	new file:   src/components/list-users/list-users.ts
	deleted:    src/components/member/index.ts
	deleted:    src/components/member/member.hbs
	deleted:    src/components/member/member.ts
	modified:   src/controllers/initial-controller.ts
	new file:   src/controllers/search-controller.ts
	modified:   src/controllers/signup-controller.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withListUsers.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/api/chats-api.ts
	modified:   src/components/list-users/list-users.hbs
	modified:   src/controllers/initial-controller.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
new file:   src/components/empty-messages/empty-messages.css
	new file:   src/components/empty-messages/empty-messages.hbs
	new file:   src/components/empty-messages/empty-messages.ts
	new file:   src/components/empty-messages/index.ts
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
new file:   src/components/modal-add-delete-user/index.ts
	new file:   src/components/modal-add-delete-user/modal-add-delete-user.css
	new file:   src/components/modal-add-delete-user/modal-add-delete-user.hbs
	new file:   src/components/modal-add-delete-user/modal-add-delete-user.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/components/dots/dots.css
	modified:   src/components/dots/dots.hbs
	modified:   src/components/dots/dots.ts
	modified:   src/components/modal-add-delete-user/modal-add-delete-user.hbs
	modified:   src/components/modal-add-delete-user/modal-add-delete-user.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withModalAddDeleteUser.ts
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/components/clip/clip.css
	modified:   src/components/clip/clip.hbs

AleksandrMenshchikov committed:
modified:   src/components/clip/clip.ts
	new file:   src/components/modal-clip/index.ts
	new file:   src/components/modal-clip/modal-clip.css
	new file:   src/components/modal-clip/modal-clip.hbs
	new file:   src/components/modal-clip/modal-clip.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withModalClip.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/components/list-users/list-users.hbs
	new file:   src/components/messages/index.ts
	new file:   src/components/messages/messages.css
	new file:   src/components/messages/messages.hbs
	new file:   src/components/messages/messages.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withMessages.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/api/chats-api.ts

AleksandrMenshchikov committed:
new file:   src/components/button-add-user/button-add-user.hbs
	new file:   src/components/button-add-user/button-add-user.ts
	new file:   src/components/button-add-user/index.ts
	new file:   src/components/button-delete-user/button-delete-user.hbs
	new file:   src/components/button-delete-user/button-delete-user.ts
	new file:   src/components/button-delete-user/index.ts
	modified:   src/components/modal-add-delete-user/modal-add-delete-user.hbs
	modified:   src/components/modal-add-delete-user/modal-add-delete-user.ts
	new file:   src/components/modal-users/index.ts
	new file:   src/components/modal-users/modal-users.css
	new file:   src/components/modal-users/modal-users.hbs
	new file:   src/components/modal-users/modal-users.ts
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/components/input-search/input-search.hbs
	new file:   src/components/list-divider/index.ts
	new file:   src/components/list-divider/list-divider.hbs
	new file:   src/components/list-divider/list-divider.ts
	modified:   src/components/list-users/list-users.css
	modified:   src/components/list-users/list-users.hbs
	modified:   src/components/list-users/list-users.ts
	modified:   src/components/modal-users/modal-users.css
	modified:   src/components/modal-users/modal-users.hbs
	modified:   src/components/modal-users/modal-users.ts
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
new file:   src/components/modal-users-title/index.ts
	new file:   src/components/modal-users-title/modal-users-title.css
	new file:   src/components/modal-users-title/modal-users-title.hbs
	new file:   src/components/modal-users-title/modal-users-title.ts
	modified:   src/components/modal-users/modal-users.css
	modified:   src/components/modal-users/modal-users.hbs
	modified:   src/components/modal-users/modal-users.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withModalUsersTitle.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
new file:   src/components/list-chats/index.ts
	renamed:    src/components/list-members/list-members.css -> src/components/list-chats/list-chats.css
	renamed:    src/components/list-members/list-members.hbs -> src/components/list-chats/list-chats.hbs
	renamed:    src/components/list-members/list-members.ts -> src/components/list-chats/list-chats.ts
	deleted:    src/components/list-members/index.ts
	modified:   src/controllers/initial-controller.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withListChats.ts
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
new file:   src/components/button-edit/button-edit.css
	new file:   src/components/button-edit/button-edit.hbs
	new file:   src/components/button-edit/button-edit.ts
	new file:   src/components/button-edit/index.ts
	modified:   src/components/button-open-profile/button-open-profile.css
	modified:   src/controllers/initial-controller.ts
	modified:   src/pages/messenger/index.css
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
new file:   src/components/modal-edit/index.ts
	new file:   src/components/modal-edit/modal-edit.css
	new file:   src/components/modal-edit/modal-edit.hbs
	new file:   src/components/modal-edit/modal-edit.ts
	modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
new file:   src/components/form-edit/form-edit.hbs
	new file:   src/components/form-edit/form-edit.ts
	new file:   src/components/form-edit/index.ts
	modified:   src/components/modal-edit/modal-edit.hbs
	modified:   src/components/modal-edit/modal-edit.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/components/modal-edit/modal-edit.ts
	new file:   src/controllers/chats-controller.ts
	modified:   src/controllers/initial-controller.ts

AleksandrMenshchikov committed:
modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/components/list-chats/list-chats.hbs
	modified:   src/components/messages/messages.hbs
	modified:   src/core/Store.ts
	modified:   src/hoc/withMessages.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/pages/messenger/index.hbs
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/components/modal-users/modal-users.ts

AleksandrMenshchikov committed:
modified:   src/api/chats-api.ts
	modified:   src/api/urls.ts
	modified:   src/controllers/chats-controller.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/api/chats-api.ts
	modified:   src/api/urls.ts
	modified:   src/controllers/chats-controller.ts
	modified:   src/core/Store.ts

AleksandrMenshchikov committed:
modified:   src/api/chats-api.ts
	modified:   src/controllers/chats-controller.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/controllers/login-controller.ts
	modified:   src/controllers/signup-controller.ts

AleksandrMenshchikov committed:
modified:   src/api/chats-api.ts
	modified:   src/api/urls.ts
	modified:   src/components/list-chats/list-chats.css
	modified:   src/components/list-chats/list-chats.hbs
	modified:   src/components/profile-arrow/profile-arrow.css
	modified:   src/components/profile-arrow/profile-arrow.hbs
	modified:   src/controllers/chats-controller.ts
	modified:   src/core/Store.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/components/clip/clip.css
	modified:   src/components/dots/dots.css
	modified:   src/components/messages/messages.css
	modified:   src/components/messages/messages.hbs
	modified:   src/components/modal-add-delete-user/modal-add-delete-user.css
	modified:   src/components/modal-clip/modal-clip.css
	modified:   src/components/profile-arrow/profile-arrow.css

AleksandrMenshchikov committed:
modified:   src/components/messages/messages.css
	modified:   src/components/messages/messages.hbs

AleksandrMenshchikov committed:
modified:   src/controllers/chats-controller.ts

AleksandrMenshchikov committed:
modified:   src/controllers/chats-controller.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
new file:   src/components/messages-content/index.ts
	new file:   src/components/messages-content/messages-content.hbs
	new file:   src/components/messages-content/messages-content.ts
	modified:   src/components/messages/messages.hbs
	modified:   src/components/messages/messages.ts
	modified:   src/controllers/chats-controller.ts
	modified:   src/core/Store.ts
	new file:   src/hoc/withMessagesContent.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/controllers/chats-controller.ts

AleksandrMenshchikov committed:
modified:   src/controllers/chats-controller.ts
	modified:   src/controllers/password-controller.ts

AleksandrMenshchikov committed:
modified:   src/controllers/chats-controller.ts
	modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   src/controllers/chats-controller.ts

AleksandrMenshchikov committed:
modified:   src/controllers/chats-controller.ts

AleksandrMenshchikov committed:
modified:   src/controllers/initial-controller.ts

gennady-bars commented on src/api/auth-api.ts:
```diff
@@ -0,0 +1,35 @@
+
+class AuthApi extends BaseAPI {
+  create(obj: Record<string, unknown>) {
+    return httpTransport.post(URLs['https://ya-praktikum.tech/api/v2/auth/signup'], {
```
`https://ya-praktikum.tech/api/v2` дублируется 38 раз в проекте

`https://ya-praktikum.tech/api/v2` везде повторяется в запросах. Нужно вынести в константу `BASE_URL` и импортировать везде в коде проекта, добавляя в конце изменяющуюся часть (эндпоит). Это обычная практика, чтобы одним движением можно было **сменить адрес**, но не менять эндпоинты.

А еще лучше поместить этот базовый урл  1 раз внутрь класса `HTTPTransport`, а в конструктор экземпляров передавать только эндпоинт

```jsx

new HTTPTransport('/auth/');

```

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on src/controllers/chats-controller.ts:
```diff
@@ -0,0 +1,225 @@
+          }).catch((err) => console.log(err));
+        }
+      })
+      .catch((err) => console.log(err));
```
Отлично, что ловите возможные ошибки в конце каждого запроса к серверу

gennady-bars commented on utils/HTTPTransport.ts:
```diff
@@ -22,26 +22,26 @@ function queryStringify(data: {[x: string]: unknown}) {
 class HTTPTransport {
-  get = (url: string, options: Options) => this.request(
-    url,
+  get = (_url: string, options: Options) => this.request(
```
**Можно лучше**
Если типизация повторяется в каждом методе, то нужно типизировать сам метод, а не дублировать типизацию аргументов. Код станет чище

```jsx

// создаем тип метода
type HTTPMethod = (url: string, options?: Options) => Promise<unknown>

export class HTTPTransport {

  // используем тип и удаляем дублирование в аргументах
  get: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.GET}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  put: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.PUT}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  post: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.POST}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  delete: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.DELETE}, options.timeout)
  )

```

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on utils/HTTPTransport.ts:
```diff
@@ -60,7 +60,7 @@ class HTTPTransport {
       const isGet = method === METHODS.GET;
 
       xhr.open(method, isGet && !!data ? `${url}${queryStringify(data)}` : url);
```
**Можно лучше**

`queryStringify` лучше сразу в методе `get` вызывать, чтобы не проверять `METHODS.GET` при каждом запросе

gennady-bars commented on src/pages/messenger/index.ts:
```diff
@@ -20,149 +93,152 @@ const location: URL = new URL(
-        if (Object.prototype.hasOwnProperty.call(this.children.members, key)) {
-          strChildren += this.children.members[key].getContent().outerHTML;
-        }
+    document.addEventListener('keydown', (e) => {
```
`  document.addEventListener('keydown', (e) => {`

**Можно лучше**

Не стоит навешивать обработчики клика на `document`, если их не удаляете. Эти обработчики срабатывают при любых кликах по сайту, а это трата ресурсов

AleksandrMenshchikov replied:
Исправил.

gennady-bars replied:
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)


К сожалению, работу придется отклонить без потери итерации, так как проект пришел без изменений. Нет коммита с изменениями на гитхабе после моего ревью

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


* `Readme` хорошо оформлен
* Отлично, что фиксируете версии зависимостей
* Отлично, что нет `EOF` ошибок в гите
* Хорошая структура папок и файлов
* Отлично, что ловите возможные ошибки в конце каждого запроса к серверу
* Отлично, то отображаете загрузку во время ожидания ответа от сервера


но есть некоторые недочеты: 


* не отображается список моих чатов и не могу создать новый. Скрин ошибок https://disk.yandex.ru/i/hnSrFOF25O_rMA 
* когда перехожу в профиль в консоли ошибки https://disk.yandex.ru/i/-J3AMkbr6C0oQA 
* не могу выйти обратно из профиля в чаты 
* если перезагрузить страницу в Нетлифай, то попадаю на 404, а роутинг должен отправлять пользователя туда, где он перезагружал страницу (если в чатах, то чаты, если в профиле - то в профиль)
* Эти все ошибки в задеплоенном приложении на Нетлифай. Нужно его обновить и  исправить
* Должна быть возможность удалить чат
* `https://ya-praktikum.tech/api/v2` дублируется 38 раз в проекте


**Можно лучше**

* Если типизация повторяется в каждом методе, то нужно типизировать сам метод, а не дублировать типизацию аргументов. Код станет чище
* Попробуйте избавиться от типов `any`
* Нужно дать возможность изменять картинку чата
* Не стоит навешивать обработчики клика на `document`, если их не удаляете. Эти обработчики срабатывают при любых кликах по сайту, а это трата ресурсов


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

AleksandrMenshchikov committed:
modified:   package-lock.json

AleksandrMenshchikov committed:
modified:   src/api/auth-api.ts
	modified:   src/api/chats-api.ts
	modified:   src/api/urls.ts
	modified:   src/api/users-api.ts
	modified:   src/controllers/avatar-controller.ts
	modified:   src/controllers/chats-controller.ts
	modified:   src/controllers/initial-controller.ts
	modified:   src/controllers/login-controller.ts
	modified:   src/controllers/profile-controller.ts
	modified:   src/controllers/search-controller.ts
	modified:   src/controllers/signup-controller.ts

AleksandrMenshchikov committed:
modified:   utils/HTTPTransport.ts

AleksandrMenshchikov committed:
modified:   src/pages/messenger/index.ts

AleksandrMenshchikov committed:
modified:   package.json

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
